<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BSTWindow</class>
 <widget class="QWidget" name="BSTWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1031</width>
    <height>667</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BSTWindow</string>
  </property>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>1011</width>
     <height>651</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_1" stretch="1,6,1">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_1" stretch="2,2,2,2,1,3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0,0">
        <item>
         <widget class="QLabel" name="label_1">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>请输入节点数N：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="getN_lineEdit"/>
        </item>
        <item>
         <widget class="QPushButton" name="createBSTButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>随机生成N个节点添加到二叉排序树</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>请输入插入的节点的值：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="insertKey_lineEdit"/>
        </item>
        <item>
         <widget class="QPushButton" name="insertButton">
          <property name="text">
           <string>插入</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>请选择节点：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="treeNode_comboBox"/>
        </item>
        <item>
         <widget class="QPushButton" name="deleteButton">
          <property name="text">
           <string>删除</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>请输入带查询节点的值：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="searchBST_lineEdit"/>
        </item>
        <item>
         <widget class="QPushButton" name="searchBST_pushButton">
          <property name="text">
           <string>查询</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>查询到的节点信息：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextBrowser" name="nodeInfo_textBrowser">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>200</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Lucida Sans Typewriter</family>
         </font>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,1,1,1,1,15">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>中序遍历序列：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextBrowser" name="inorder_textBrowser">
        <property name="font">
         <font>
          <family>Lucida Sans Typewriter</family>
         </font>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>逆序的中序遍历序列：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextBrowser" name="reverseInorder_textBrowser">
        <property name="font">
         <font>
          <family>Lucida Sans Typewriter</family>
         </font>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>树状表示图：</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextBrowser" name="printTree_textBrowser">
        <property name="font">
         <font>
          <family>Lucida Sans Typewriter</family>
         </font>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QPushButton" name="clearTree_pushButton">
        <property name="text">
         <string>清空二叉排序树</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="returnButton">
        <property name="text">
         <string>返回</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
